src/main/java/org/folio/search/integration/KafkaMessageListener.java

```diff
@@ -212,6 +212,22 @@ public void handleBibframeEvents(List<ConsumerRecord<String, ResourceEvent>> con
     indexResources(batch, resourceService::indexResources);
   }
 
+  @KafkaListener(
+    id = KafkaConstants.CAMPUS_LISTENER_ID,
+    containerFactory = "standardListenerContainerFactory",
+    groupId = "#{folioKafkaProperties.listener['campus'].groupId}",
+    concurrency = "#{folioKafkaProperties.listener['campus'].concurrency}",
+    topicPattern = "#{folioKafkaProperties.listener['campus'].topicPattern}")
+  public void handleCampusEvents(List<ConsumerRecord<String, ResourceEvent>> consumerRecords) {
+    log.info("Processing campus events from Kafka [number of events: {}]", consumerRecords.size());
+    var batch = consumerRecords.stream()
+      .map(ConsumerRecord::value)
+      .map(campus -> campus.id(getResourceEventId(campus) + "|" + campus.getTenant()))
+      .toList();
+
+    indexResources(batch, resourceService::indexResources);
+  }
+
```

Comment:
I think there is no sense in creating a new listener for location unit events. It could be combined with location listener. It could be achieved by modifying `topic-pattern` property to subscribe to both topics. Like: `(${folio.environment}\.)(.*\.)inventory\.(location|campus)`

Reply:
It was fixed, @psmagin please take a look again